rllib/env/multi_agent_episode.py

```diff
@@ -0,0 +1,535 @@
+import uuid
+
+from typing import Any, Dict, List, Optional, Union
+
+from ray.rllib.env.single_agent_episode import SingleAgentEpisode
+from ray.rllib.policy.sample_batch import MultiAgentBatch
+
+
+class IndexMapping(list):
+    """Provides lists with a method to find elements."""
```

Comment:
Can you explain more what this will be used for?

Plus add docstring to the `find_indices` method.